src/Features/Core/Portable/CodeRefactorings/IRefactoringHelpersService.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings
+{
+    interface IRefactoringHelpersService : ILanguageService
```

Comment:
There's a service with similar name (and very different usecase) called `ICodeRefactoringHelpersService` with `bool ActiveInlineRenameSession { get; }`. We might consider using different name, maybe something more direct `INodeForSpanHelperService` or `INodeExtractorHelperService`...

Reply:
I am fine with any name here given we are putting it in a very specific namespace. Can we add some doc comments on the service to clarify the intent?

Reply:
Same here, please add `internal` modifier.

Reply:
As for the internal modifier since refactorings can be authored by 3rd parties as well, don't we want to surface helpers for them as well?

Reply:
As for docs, yep, I'll have to rewrite them completely.

Reply:
> As for the internal modifier since refactorings can be authored by 3rd parties as well, don't we want to surface helpers for them as well

Yes, but that would need to go through a more rigorous design review if we want to expose public APIs for these. I would recommend keeping them internal for now, and track making these public as a separate exercise.